tls/ttls.c

```diff
@@ -164,6 +164,7 @@ ttls_crypto_req_sglist(TlsCtx *tls, struct crypto_aead *tfm, unsigned int len,
 			n = skb_to_sgvec(skb, sg_i, off, to_read);
 			if (n <= 0)
 				goto err;
+			sg_unmark_end(sg_i + n - 1);
```

Comment:
The statement doesn't seem correct. If n is the same as previous and we're going to exit on current iteration from the loop, then we lose just set mark end by `skb_to_sgvec()`. I think `sg_unmark_end()` must be just before `skb_to_sgvec()` call to unset previously set end mark (either from `sg_init_table()` or previous iteration) and let `skb_to_sgvec()` define the last (current) end mark.

Reply:
I believe this is not true, and the proposed version is fine. As there is an explicit call of  `sg_mark_end(*sg + *sgn - 1)` just outside the loop, we will never fail to put the end mark.

Although it's possible to remove that `sg_mark_end()` and to call `sg_unmark_end()` only when it's absolutely necessary, it will make code look wrong at the first glance as there will be no explicit `sg_mark_end()`. Then, reader will realize that it's `skb_to_sgvec()` that sets the mark. And it will gain nothing, as the saved load-store in (inlined) `sg_unmark_end()` will be eaten by the tracking logic — one must avoid calling `sg_unmark_end()` on the first iteration. I'd like propose to leave the calls as they are now.

Still, if you prefer to have `skb_to_sgvec()` setting the last mark, I'll change the code to do that.